### ISSUE TITLE
Do not push a downloaded update in case of regression

### ISSUE THREAD

**Issue by biva** [nextcloud/desktop#4044]
### Feature description
Following https://github.com/nextcloud/desktop/issues/4016#issuecomment-987013976 it would be great to _not_ push the update in case of a regression for example.

### Steps
1. Download an update for a new version (v10)
2. Install it
3. See that there's a regression
4. Install the previous stable version available (v9)

### What is happening now
5. The automatic update for (v10) is proposed to the user

### What should happen
5. The automatic update (v10) is not proposed until an official new version (v11, correcting the bug) is pushed by Nextcloud.
 
In other words, if a regression is discovered in v10, Nextcloud removes if from download and from automatic update. The automatic update will be pushed only for v11. But if the file had been previously downloaded, the client keeps pushing for v10, while it shouldn't.

**Comment by dave-kennedy** [comment]
This isn't a bad idea but I'd rather see a full suite of tests to make sure major regressions don't make it to release in the first place, including:

1. As close to 100% unit test coverage as is possible and reasonable
2. Automated integration/end-to-end tests - the distinction between this kind of test and unit tests is, while unit tests normally mock external dependencies, these do not
3. Manual testing - double check critical functionality with physical eyeballs because automated tests are no substitute for actual humans
